dev/TeachingTip/TeachingTip.cpp

```diff
@@ -1119,12 +1120,29 @@ void TeachingTip::OnF6AcceleratorKeyClicked(const winrt::CoreDispatcher&, const
 
             if (f6Button)
             {
-                auto const scopedRevoker = f6Button.GettingFocus(winrt::auto_revoke, [this](auto const&, auto const& args) {
+                auto const scopedRevoker = f6Button.GettingFocus(winrt::auto_revoke, [this](auto const&, const winrt::GettingFocusEventArgs& args) {
                     m_previouslyFocusedElement = winrt::make_weak(args.OldFocusedElement());
                 });
                 const bool setFocus = f6Button.Focus(winrt::FocusState::Keyboard);
                 args.Handled(setFocus);
             }
+            // If there is no close button, try setting focus on the tip's custom content.
+            else if (const auto mainContentPresenter = m_mainContentPresenter.get())
+            {
+                if (const auto focusableElement = winrt::FocusManager::FindFirstFocusableElement(mainContentPresenter))
+                {
+                    // A focusable element can be a Hyperlink which does not expose a GettingFocus event. As such, we use the FocusManager API here. 
+                    auto const scopedRevoker = winrt::FocusManager::GettingFocus(winrt::auto_revoke, [this, focusableElement](auto const&, const winrt::GettingFocusEventArgs& args) {
```

Comment:
Looks like I missed the fact that the [FocusManager::GettingFocus](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.input.focusmanager.gettingfocus?view=winrt-19041) API is only available starting with RS5 (hence the test failures on RS4 and below).

For now, we only enter this case (no close button) if the tip is light-dismissable. In that case, the previously focused element (once the focusable custom content element got focus) is the tip's internal light-dismiss Popup control. As such, we don't actually need the to listen to the GettingFocus event here, since the previously focused element is always the same and one we know - teaching tip's internal `m_lightDismissIndicatorPopup`.

As such, I think I can rewrite the code above like this:
```cpp
// If there is no close button, try setting focus on the tip's custom content.
else if (const auto mainContentPresenter = m_mainContentPresenter.get())
{
    if (const auto focusableElement = winrt::FocusManager::FindFirstFocusableElement(mainContentPresenter))
    {
        const bool setFocus = SetFocus(focusableElement, winrt::FocusState::Keyboard);
        if (setFocus)
        {
            if (const auto lightDismissPopupAsDO = m_lightDismissIndicatorPopup.get().try_as<winrt::DependencyObject>())
            {
                m_previouslyFocusedElement = winrt::make_weak(lightDismissPopupAsDO);
            }
        }
        args.Handled(setFocus);
    }
}
```

@StephenLPeters @ranjeshj Your thoughts?

Reply:
On another note, I have also noticed that pressing F6 to put focus on a tip in a Xaml Island scenario does *not* work (no matter if focue is already in the island or outside of it at the time of F6). Is that a known issue? 

One implemention issue I see with Xaml island support is that we expect our focusable elements to be of type Windows.UI.Xaml.DependencyObject, yet, for example, a WPF button is of type System.Windows.DependencyObject.

So... is the current guidance that developers using a tip in a Xaml island need to take care of keyboard accessibility (F6) on their own? And if so, we should update the teaching tip documentation to reflect that.

Reply:
@JesseCol @Austin-Lamb for the Xaml Island behavior. I would like to avoid controls having to do special behavior for Islands as much as possible.